### PR TITLE
CI: Check and fix missing __init__.py with flake8-no-pep420

### PR DESCRIPTION
**Pull request by intgr**
## Related issues

As suggested in https://github.com/typeddjango/django-stubs/pull/1460#issuecomment-1525875248

### CODE REVIEW

**Comment by adamchainz** [comment]
Ha, I was just working on this, and adding `.pyi` support to the plugin: https://github.com/adamchainz/flake8-no-pep420/pull/183

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -34,6 +34,7 @@ repos:
     hooks:
       - id: flake8
         additional_dependencies:
+        - flake8-no-pep420==2.3.0
```
Comment (adamchainz) [inline .pre-commit-config.yaml]: The release is ready!

```suggestion
        - flake8-no-pep420==2.4.0
```

Finds these errors:

```
django-stubs/contrib/postgres/forms/jsonb.py:1:1: INP001 File is part of an implicit namespace package. Add an __init__.py?
django_stubs_ext/django_stubs_ext/db/models.py:1:1: INP001 File is part of an implicit namespace package. Add an __init__.py?
django_stubs_ext/setup.py:1:1: INP001 File is part of an implicit namespace package. Add an __init__.py?
django_stubs_ext/tests/test_aliases.py:1:1: INP001 File is part of an implicit namespace package. Add an __init__.py?
django_stubs_ext/tests/test_monkeypatching.py:1:1: INP001 File is part of an implicit namespace package. Add an __init__.py?
setup.py:1:1: INP001 File is part of an implicit namespace package. Add an __init__.py?
tests/test_error_handling.py:1:1: INP001 File is part of an implicit namespace package. Add an __init__.py?
```

`setup.py` can be ignored, but I think the others are worth adding the `__init__.py` files for.

Comment (christianbundy) [reply .pre-commit-config.yaml]: >The release is ready!

so fast! 🔥

**Comment by intgr** [comment]
ISTM `tests` isn't really a package, so I wouldn't add `__init__.py` there?

Also I renamed `django-stubs/contrib/postgres/forms/jsonb.py` to `.pyi`, this was a mistake clearly.

**Review by sobolevn — APPROVED** [review]
The next step is to "steal" some checks from `typeshed` to catch `.py` file extensions 😆 

See https://github.com/python/typeshed/blob/main/tests/check_consistent.py

@adamchainz you made so many awesome things, thank you 👍

**Comment by adamchainz** [comment]
> ISTM `tests` isn't really a package, so I wouldn't add `__init__.py` there?

Eh, I normally make my `tests` folder a package, but it's a bit different here since all the tests are yaml files.

> Also I renamed django-stubs/contrib/postgres/forms/jsonb.py to .pyi, this was a mistake clearly.

Right you are!

> @adamchainz you made so many awesome things, thank you 👍

Just solving problems I encounter, like yourself! 🫡

**Comment by intgr** [comment]
> Eh, I normally make my tests folder a package, but it's a bit different here since all the tests are yaml files.

Actually you're right. It makes sense that they're a packge, because a test module may want to import code from other tests modules.

**Comment by intgr** [comment]
Hrm, adding `__init__.py` to `tests` [causes errors in CI](https://github.com/typeddjango/django-stubs/actions/runs/4830749225/jobs/8607333959?pr=1461):

```
ImportError while importing test module '/home/runner/work/django-stubs/django-stubs/django_stubs_ext/tests/test_aliases.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
E   ModuleNotFoundError: No module named 'tests.test_aliases'
```

Looks like it causes `tests` and `django_stubs_ext/tests` to conflict.

Are there any solutions or should I just revert these `__init__.py`? I have no strong opinion on this.

**Comment by sobolevn** [comment]
I never use `__init__.py` in tests.

**Pull request merged**